crime-applications-adaptor/src/main/java/uk/gov/justice/laa/crime/applications/adaptor/mapper/CrimeApplyMapper.java

```diff
@@ -0,0 +1,96 @@
+        crimeApplication.setDateCreated(crimeApplyResponse.getSubmittedAt());
+        crimeApplication.setDateStamp(crimeApplyResponse.getDateStamp());
+        crimeApplication.setApplicant(mapApplicant(crimeApplyResponse));
+
+        return crimeApplication;
+    }
+
+    private String mapSolicitorName(Provider providerDetails) {
+        if (providerDetails == null) {
+            return null;
+        }
+
+        List<String> firstAndLastName = List.of(providerDetails.getLegalRepFirstName(),
+                providerDetails.getLegalRepLastName());
+        return String.join(StringUtils.SPACE, firstAndLastName);
+    }
+
+    private Applicant mapApplicant(MaatApplication crimeApplyResponse) {
+        Applicant applicant = new Applicant();
+
+        uk.gov.justice.laa.crime.applications.adaptor.model.criminalapplicationsdatastore.Applicant crimeApplyApplicant =
+                crimeApplyResponse.getClientDetails().getApplicant();
+
+        applicant.setFirstName(crimeApplyApplicant.getFirstName());
+        applicant.setOtherNames(crimeApplyApplicant.getOtherNames());
+        applicant.setSurname(crimeApplyApplicant.getLastName());
+        applicant.setDateOfBirth(crimeApplyApplicant.getDateOfBirth());
+        applicant.setTelephone(crimeApplyApplicant.getTelephoneNumber());
+        applicant.setNiNumber(crimeApplyApplicant.getNino());
+        applicant.setNoFixedAbode(mapNoFixedAbode(crimeApplyApplicant.getHomeAddress()));
+        applicant.setUseSupplierAddressForPost(mapUseSupplierAddressForPost(crimeApplyApplicant));
+        applicant.setHomeAddress(mapAddress(crimeApplyApplicant.getHomeAddress()));
+        applicant.setPostalAddress(mapAddress(crimeApplyApplicant.getCorrespondenceAddress()));
+
+        return applicant;
+    }
+
+    private Boolean mapUseSupplierAddressForPost(uk.gov.justice.laa.crime.applications.adaptor.model.criminalapplicationsdatastore.Applicant crimeApplyApplicant) {
+        uk.gov.justice.laa.crime.applications.adaptor.model.criminalapplicationsdatastore.Applicant.CorrespondenceAddressType addressType =
+                crimeApplyApplicant.getCorrespondenceAddressType();
+        switch (addressType) {
+            case OTHER_ADDRESS, HOME_ADDRESS -> {
+                return false;
+            }
+            case PROVIDERS_OFFICE_ADDRESS -> {
+                return true;
+            }
+            default -> throw new IllegalStateException("Unexpected value: " + addressType);
+        }
+    }
+
+    private boolean mapNoFixedAbode(uk.gov.justice.laa.crime.applications.adaptor.model.criminalapplicationsdatastore.general.Address homeAddress) {
+        return homeAddress == null;
+    }
+
+    private Address mapAddress(uk.gov.justice.laa.crime.applications.adaptor.model.criminalapplicationsdatastore.general.Address crimeApplyAddress) {
+        if (crimeApplyAddress == null) {
+            return null;
+        }
+        Address Address = new Address();
```

Comment:
please change Address to address, it is code smell in sonarqube. thanks

Reply:
done and pushed